tasks/init.js

```diff
@@ -35,83 +35,202 @@ var config = defaults.getInstance('init')
+            return !!(opt.value[selType]);
+          });
+        if (!opt.value.isMultiple) {
+          validTypes.forEach(function(validType) {
+            if (valueType === validType) {
+              typeIsOk = true;
+            }
+          });
         }
-      });
-    },
-    port: function (value) {
-      switch (typeof value) {
-        case 'number':
-          if (isNaN(parseInt(value))) {
-            return 'port must be integer';
+        else {
+          value = [].concat(value);
+          if (value.length === 0) {
+            typeIsOk = true;
           }
-          break;
-        case 'string':
-          if (value === 'random') break;
-        default:
-          return 'port must be an integer or the keyword "random"';
-      }
-    },
-    npm : function (value) {
-      if (typeof value !== 'boolean') {
-        return 'bower must be a boolean';
-      }
-    },
-    karma : function (value) {
-      if (typeof value !== 'boolean') {
-        return 'karma must be a boolean';
-      }
-    },
-    jshint : function (value) {
-      if (typeof value !== 'boolean') {
-        return 'jshint must be a boolean';
+          else {
+            value.forEach(function (subValue) {
+              validTypes.forEach(function(validType) {
+                if (typeof subValue === validType) {
+                  typeIsOk = true;
+                }
+              });
+            });
+          }
+        }
+        if (!typeIsOk) {
+          var expectedType = JSON.stringify(validTypes);
+          if (opt.value.isMultiple) {
+            expectedType = 'Array<' + expectedType + '>'
+          }
+          throw new Error('' + key + ' is expected to be of types ' +
+            expectedType + ' but a ' + (typeof value) + ' was provided instead.');
+        }
       }
-    },
-    gitignore : function (value) {
-      if (typeof value !== 'boolean') {
-        return 'gitignore must be a boolean';
+
+      if (key === 'version') {
+        // version needs an additional test than the default one for a string
+        if (!( /\d+\.\d+\.\d+[-\w\d]*/ ).test(value)) {
+          throw new Error('version must be 3-term semver string');
+        }
       }
-    },
-    editorconfig : function (value) {
-      if (typeof value !== 'boolean') {
-        return 'editorconfig must be a boolean';
+      else if (key === 'port') {
+        // port gets special treatment because it is valid with more than one type
+        if ((valueType === 'number' && isNaN(parseInt(value))) ||
+          (valueType === 'string' && value !== 'random') ||
+          (valueType !== 'string' && valueType !== 'number')) {
+          console.log('port', value, 'valueType', valueType);
+          throw new Error('Port must be an integer, or "random"');
+        }
       }
-    }
-  })
-  .commit();
+    });
+    return true;
+  }
+}
 
-yargs.getInstance('init')
-  .usage(wordwrap(2, 80)([
+taskYargs.register('init', {
+  description: (wordwrap(2, 80)([
```

Comment:
The wordwrap() is a fix for a yargs bug, can it be encapsulated in the third party library?